filament/src/Renderer.cpp

```diff
@@ -259,7 +256,21 @@ void FRenderer::renderJob(ArenaScope& arena, FView& view) {
 
     FrameGraph fg(engine.getResourceAllocator());
 
-    const TextureFormat hdrFormat = getHdrFormat(view);
+    // Figure out if we need to blend this view into the framebuffer. Maybe this should be
+    // explicit, but since we don't have an API right now, we use heuristics:
+    // - we are keeping the color buffer before rendering, and
+    // - we are not clearing or clearing with alpha
+    // FIXME: make this an explicit API
+    const bool blending = !(view.getDiscardedTargetBuffers() & TargetBufferFlags::COLOR)
```

Comment:
We do have an explicit API for transparent rendering, it's a flag on the SwapChain. We should also check for clearColor.a < 1.0 even if the flag is set though.

Reply:
Actually the swapchain API only tells us that the buffer we're writing into is transparent, not that we should bend into it (which involves a full buffer blend) -- e.g. you might just want to clear the view with 0,0,0,0 and then draw opaque things in it.

Conversely, you could have an opaque swapchain and want to draw several views in it on top of each other, with blending.

So we need to know for sure that the user intended to blend the view as a whole.  Currently, I'm using the state of the clearcolor + discardflags to infer that.

The clearColor alpha check is done just one line below (line 265).

Reply:
Ah right I forgot about the use case of views blended on top of an opaque framebuffer. I saw the check for clearColor, that's why I mentioned it when talking about the flag.